src/framework/components/element/component.js

```diff
@@ -734,24 +752,62 @@ pc.extend(pc, function () {
         // internal set width without updating margin
         _setWidth: function (w) {
             this._width = w;
-
-            var i,l;
-            var c = this.entity._children;
-            for (i = 0, l = c.length; i < l; i++) {
-                if (c[i].element) {
-                    c[i].element._anchorDirty = true;
-                    c[i].element._sizeDirty = true;
-                }
-            }
+            this._setCalculatedWidth(w, false);
 
             this.fire('set:width', this._width);
-            this.fire('resize', this._width, this._height);
         },
 
         // internal set height without updating margin
         _setHeight: function (h) {
             this._height = h;
+            this._setCalculatedHeight(h, false);
+
+            this.fire('set:height', this._height);
+        },
+
+        _setCalculatedWidth: function(value, updateMargins) {
+            var didChange = Math.abs(value - this._calculatedWidth) > 1e-4;
+
+            this._calculatedWidth = value;
 
+            if (updateMargins) {
+                var p = this.entity.getLocalPosition().data;
+                var pvt = this._pivot.data;
+                this._margin.data[0] = p[0] - this._calculatedWidth * pvt[0];
+                this._margin.data[2] = (this._localAnchor.data[2] - this._localAnchor.data[0]) - this._calculatedWidth - this._margin.data[0];
+            }
+
+            this._flagChildrenAsDirty();
+
+            this.fire('set:calculatedWidth', this._calculatedWidth);
+
+            if (didChange) {
+                this.fire('resize', this._calculatedWidth, this._calculatedHeight);
+            }
```

Comment:
This change is potentially significant – it's designed to stop the `resize` event from being fired if a resize didn't really take place. Hopefully people don't already depend on receiving a `resize` event in the case where there wasn't actually a resize, but they could do. We could make this conditional perhaps, or do it on the recipient's end by including the previous `width` and `height` values in the event payload.

Reply:
I think that's probably fine